README.md

```diff
@@ -31,8 +31,13 @@ TODO
 All examples assume a client and/or channel has been created as follows:
 
 ```go
-client := ably.NewRestClient(ably.ClientOptions{ApiKey: "xxxx"})
-channel := client.Channel('test')
+options, err := ably.NewClientOptions("xxx:xxx")
```

Comment:
Yes, I wanted to do that in the original version, but in order to make that possible, I needed to have meaningful zero-value for ably.RestClient, so the factory method could take the key string intead of options (Go has no method overloading). The other options is to have to factory methods, one NewRestClient(key string) and second NewRestClientOptions(*CientOptions). If you like the latter idea, I can make it.

The factory method for ClientOptions returns error because it's parsing argument, and parsing can fail.